ligo/stablecoin/actions.ligo

```diff
@@ -240,4 +338,237 @@ function is_operator_action
+  ( const parameters : mint_params
+  ; const store      : storage
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_minter (store)
+
+; const senderAddress : address = Tezos.sender
+
+; function mint_tokens
+    ( const accumulator       : storage
+    ; const mint_param        : mint_param
+    ; const current_allowance : nat
+    ) : storage is block
+  { const unwrapped_parameter : mint_param_ =
+      Layout.convert_from_right_comb ((mint_param : mint_param))
+  ; fail_on
+      ( unwrapped_parameter.amount > current_allowance
+      , "ALLOWANCE_EXCEEDED"
+      )
+
+  ; const updated_allowance : nat =
+      abs (current_allowance - unwrapped_parameter.amount)
+  ; const updated_allowances : minting_allowances =
+      Big_map.update (Tezos.sender, Some (updated_allowance), accumulator.minting_allowances)
+  ; const updated_store : storage = accumulator with record
+      [ minting_allowances = updated_allowances
+      ; total_supply = accumulator.total_supply + unwrapped_parameter.amount
+      ]
+  } with credit_to (unwrapped_parameter, updated_store)
+
+} with
+  ( (nil : list (operation))
+  , List.fold
+      ( function
+          ( const accumulator : storage
+          ; const mint_param  : mint_param
+          ) : storage is case accumulator.minting_allowances[senderAddress] of
+            None -> (failwith ("NOT_MINTER") : storage)
+          | Some (current_allowance) ->
+              mint_tokens (accumulator, mint_param, current_allowance)
+          end
+      , parameters
+      , store
+      )
+  )
+
+(*
+ * Decreases balance and total supply of tokens by the given amount
+ *)
+function burn
+  ( const parameters : burn_params
+  ; const store      : storage
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_minter (store)
+; const sender_address : address = Tezos.sender
+; function burn_tokens
+    ( const accumulator : storage
+    ; const burn_amount : nat
+    ) : storage is block
+  { const debited_storage : storage = debit_from
+      ( record
+          [ from_ = sender_address
+          ; amount = burn_amount
+          ]
+      , accumulator
+      )
+  } with debited_storage with record
+      [ total_supply = case is_nat (debited_storage.total_supply - burn_amount) of
+          Some (nat) -> nat
+        | None -> (failwith ("NEGATIVE_TOTAL_SUPPLY") : nat)
+        end
+      ]
+} with
+  ( (nil : list (operation))
+  , List.fold (burn_tokens, parameters, store)
+  )
+
+(*
+ * Sets contract owner to a new address
+ *)
+function transfer_ownership
+  ( const parameter : transfer_ownership_param
+  ; const store     : storage
+  ) : entrypoint is block
+{ authorize_owner (store)
+} with
+  ( (nil : list (operation))
+  , store with record [roles.pending_owner = Some (parameter)]
+  )
+
+(*
+ * When pending sender calls this entrypoint, the address
+ * receives owner privileges for contract
+ *)
+function accept_ownership
+  ( const parameter : accept_ownership_param
```

Comment:
`accept_ownership_param` is `unit`. Isn't it more efficient to not pass it here at all?

Reply:
It's stripped anyway 🤷

Reply:
So will this function have type `lambda storage foo`? Not `lambda (pair unit storage) foo` (for some `foo`)? Did you check that?

Reply:
Yes, it will be inlined